### PR TITLE
Extends default docs setup

### LINKED ISSUE #2: Add template for developer docs
Add a template that includes docs for developers (could be in readme) as markdown page in the `docs/`. Should include things such as:

- pre-commit
- editable install
- running tests
- building docs
- ...

### PR DESCRIPTION
**Pull request by SimonHeybrock**
Copied a bunch from Scitacean.

Closes #2.

### CODE REVIEW

File: docs/developer/dependency-management.md.jinja
Comment (jl-wynen) [inline docs/developer/dependency-management.md.jinja:1]: Copied from Scitacean? :P

Note that Scipp also has this page: https://scipp.github.io/reference/developer/dependencies.html

Seems like it should be merged with this one here.

File: requirements/docs.in [hunk 1]
```diff
@@ -6,6 +6,6 @@ myst-parser
 nbsphinx
 pydata-sphinx-theme>=0.13
 sphinx
-sphinx-autodoc-typehints
+sphinx-autodoc-typehints==1.23.0  # Higher versions require sphinx-7, which is not supported by sphinx-design yet
```
Comment (jl-wynen) [inline requirements/docs.in:9]: Do we actually use sphinx-design in Scipp?

File: docs/developer/getting-started.md.jinja
[the remark this answers is not in the document]
Comment (SimonHeybrock) [reply docs/developer/getting-started.md.jinja:1]: Not sure we will ever use this for Scipp?

File: docs/developer/dependency-management.md.jinja
Comment (SimonHeybrock) [reply docs/developer/dependency-management.md.jinja:1]: > Copied from Scitacean? :P

Yes, see PR message?

File: requirements/docs.in (hunk 1 above)
Comment (SimonHeybrock) [reply requirements/docs.in:9]: This is not for Scipp. Is it needed for docs like Scitacean's?

File: docs/developer/getting-started.md.jinja
[the remark this answers is not in the document]
Comment (jl-wynen) [reply docs/developer/getting-started.md.jinja:1]: We do have a number of common files especially things related to docs. But also the pre-commit setup. So I was hoping we could use the templates for them but have a bunch of specialised files that are skipped by copier.

File: requirements/docs.in (hunk 1 above)
Comment (jl-wynen) [reply requirements/docs.in:9]: It is needed for tabs like here: https://scicatproject.github.io/scitacean/user-guide/installation.html But not for anything else in Scitacean.

And you added something similar in Markdown in this PR. I don't know if this is based on sphinx-design.

File: docs/developer/getting-started.md.jinja
[the remark this answers is not in the document]
Comment (SimonHeybrock) [reply docs/developer/getting-started.md.jinja:1]: We can look into that once we actually move Scipp to the new theme?

**Pull request merged**